packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -263,3 +327,62 @@ export function evalBlobsAndTrees(snapshot: IOdspSnapshot) {
     }
     return { numTrees, numBlobs, encodedBlobsSize, decodedBlobsSize };
 }
+
+export function isRedeemSharingLinkError(odspResolvedUrl: IOdspResolvedUrl, error: any) {
+    const errorType = error.errorType;
+    if (odspResolvedUrl.sharingLinkToRedeem !== undefined
+        && (errorType === DriverErrorType.authorizationError
+        || errorType === DriverErrorType.fileNotFoundOrAccessDeniedError)) {
+        return true;
+    }
+    return false;
+}
+
+function encodeShareUrl(url: string): string | undefined {
+    if (!url) {
+      return undefined;
+    }
+
+    /**
+     * Encode the url to accepted format by Sharepoint
+     * https://docs.microsoft.com/en-us/onedrive/developer/rest-api/api/shares_get
+     */
+    let encodedUrl = b64EncodeUnicode(url);
+    if (!encodedUrl) {
```

Comment:
I do not see a need to have this check. If you want to have it, convert to assert